### PR TITLE
Drop notifications table

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
Removes Notifications model and drops corresponding table. 

## WHY
The table is not used. 

## HOW
The supporting code has already been removed in #8370
the migration took 15 seconds on staging, which was almost the same table size as prod.

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Remove-Notifications-Table-cff7996900d84a68a39e65ab2a4cd416

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  I have deleted tests.
Have you deployed to Staging? | about to
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | yes

### CODE REVIEW

File: services/QuillLMS/db/structure.sql [hunk 1]
```diff
@@ -69,6 +69,18 @@ CREATE FUNCTION public.blog_posts_search_trigger() RETURNS trigger
       $$;
 
 
+--
+-- Name: my_jsonb_to_hstore(jsonb); Type: FUNCTION; Schema: public; Owner: -
+--
+
+CREATE FUNCTION public.my_jsonb_to_hstore(jsonb) RETURNS public.hstore
```
Comment (happythenewsad) [inline services/QuillLMS/db/structure.sql:76]: most of the changes in this file are not related to my change. presumably they were not checked in when their corresponding migrations were run.

**Review by cissyxyu — APPROVED** [review]

**Review by anathomical — APPROVED** [review]
